### PR TITLE
fixes for VFIO_USER_DIRTY_PAGES

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

**Comment by jlevon** [comment]
This is still a draft - I have to merge with Thanos's unmap changes - but it's still mostly ready for review, thanks.

**Comment by jlevon** [comment]
Fixes #533

**Review by tmakatos — APPROVED** [review]
Looks good, minor nits. I haven't looked at the unit tests very carefully.

**Review by tmakatos — COMMENTED** [review]
LGTM, minor nits.

**Comment by tmakatos** [comment]
Once this is merged we should ping JJ to make sure QEMU has the necessary changes.

File: docs/vfio-user.rst [hunk 1]
```diff
@@ -1543,16 +1543,44 @@ VFIO Dirty Bitmap Get Format
 * *size* is the size of the IOVA region
 
 * *bitmap* is the VFIO bitmap (``struct vfio_bitmap``). This field is explained
-  in `VFIO bitmap format`_.
+  in `VFIO Bitmap Format`_.
+
+This part of the request is analogous to vfio's ``struct
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/vfio-user.rst:1553]: I checked, and the rendering looked fine.

File: docs/vfio-user.rst [hunk 2]
```diff
@@ -1543,16 +1543,44 @@ VFIO Dirty Bitmap Get Format
...
++--------------+--------+-----------------------------------------+
+| bitmap range | 8      | 40                                      |
++--------------+--------+-----------------------------------------+
+| bitmap       | 48     | variable                                |
++--------------+--------+-----------------------------------------+
+
+* *argsz* is the size required for the full reply payload (dirty pages structure
+  + bitmap range structure + actual bitmap)
+* *flags* is ``VFIO_IOMMU_DIRTY_PAGES_FLAG_GET_BITMAP``
+  plus the size of any region capabilities)
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/vfio-user.rst:1585]: oops, need to remove the whole line

**Review by tmakatos — APPROVED** [review]

**Pull request merged**